jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1488,13 +1489,33 @@ Method getMethodHelper(
 			return null;
 		}
 	}
-	result = forDeclaredMethod ? getDeclaredMethodImpl(name, parameterTypes, strSig, null) : getMethodImpl(name, parameterTypes, strSig);
+	
+	if (forDeclaredMethod) {
+		result = getDeclaredMethodImpl(name, parameterTypes, strSig, null);
+	} else {
+		result = getMethodImpl(name, parameterTypes, strSig);
+		/* Retrieve the specified method implemented by the superclass from the top to the bottom
+		 * Note: there is no need do so when the method is declared by the current class.
+		 */
+		if ((result != null) && result.getDeclaringClass().isInterface() && (this != Object.class)) {
+			if (!this.isInterface()) {
+				HashSet<Class<?>> interfaceSet = new HashSet();
+				result = getMostSpecificMethodFromAllInterfacesOfAllSuperclasses(this, interfaceSet, name, strSig, parameterTypes);
+				candidateFromInteface = true;
+			} else if (result.getDeclaringClass() != this) { /* only applies if resulting class is not the base class */
+				HashSet<Class<?>> interfaceSet = new HashSet();
+				result = getMostSpecificMethodFromAllInterfacesOfCurrentClass(this, interfaceSet, null, name, strSig, parameterTypes);
```

Comment:
This method might return `null` while initial `result` isn't, this doesn't seem right.